### PR TITLE
[NO-JIRA] Infinite scroll docs

### PR DESCRIPTION
**Pull request by tiagoengel**
0 0 0 0_8080_components_infinite-scroll_platform web 1

### CODE REVIEW

File: packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  'Item 17',
+  'Item 18',
+  'Item 19',
+  'Item 20',
+];
+
+const List = ({ elements }) => (
+  <div className={getClassName('bpk-docs-infinite-scroll-page__list')}>
+    {elements.map(element => <div key={element}>{element}</div>)}
+  </div>
```
Comment (shaundon) [inline packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js]: It might be better to dogfood our own components and use `BpkList` and `BpkListItem` in the examples here.

File: packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+);
+
+const InfiniteList = withInfiniteScroll(List);
+const defaultFetch = onItemsFetch(elementsArray);
+
+const components = [
+  {
+    id: 'infinite',
+    title: 'Infinite list',
```
Comment (shaundon) [inline packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js]: This should probably be `default`.

File: packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    blurb: [],
```
Comment (shaundon) [inline packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js]: I think the examples on this page could use some blurb, as it's not entirely clear from the page what they do. @jamesf3rguson may have thoughts on what to put, but I was thinking something along the lines of:

> Loads five more list items upon reaching the bottom of the list.

File: packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    title: 'One element per scroll',
+    blurb: [],
+    examples: [
+      <FixedHeightPanel>
+        <InfiniteList onItemsFetch={defaultFetch} elementsPerScroll={1} />
+      </FixedHeightPanel>,
+    ],
+  },
+  {
+    id: 'infiniteStop',
```
Comment (shaundon) [inline packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js]: Our convention here is to use kebab case, e.g. `infinite-stop`.

File: packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+            })
+          }
+        />
+      </FixedHeightPanel>,
+    ],
+  },
+  {
+    id: 'infinite1',
+    title: 'One element per scroll',
+    blurb: [],
```
Comment (shaundon) [inline packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js]: Blurb here would be good too I think, something like:

> Loads one more list item upon reaching the bottom of the list.

File: packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    examples: [
+      <FixedHeightPanel>
+        <InfiniteList onItemsFetch={defaultFetch} elementsPerScroll={1} />
+      </FixedHeightPanel>,
+    ],
+  },
+  {
+    id: 'infiniteStop',
+    title: 'Partial load',
```
Comment (shaundon) [inline packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js]: I think a blurb is needed here, something like:

> After 15 items have loaded, a button must be pressed to load more items.

File: packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+          elementsPerScroll={5}
+          seeMoreAfter={2}
+          renderSeeMoreComponent={CustomSeeMore}
+        />
+      </FixedHeightPanel>,
+    ],
+  },
+  {
+    id: 'infiniteCustomLoading',
+    title: 'Using a custom loadingComponent',
```
Comment (shaundon) [inline packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js]: _Using a custom loading component_ would be better title, we want to keep it in plain English.

File: packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.scss [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@import '~bpk-mixins';
+
+.bpk-docs-infinite-scroll-page {
+  &__fixed-panel {
+    height: 300px; /* stylelint-disable-line unit-blacklist */
```
Comment (shaundon) [inline packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.scss]: We should use a token here, just take one of the existing ones and multiply it to make something of a similar height. E.g. `$bpk-spacing-lg * 10`

File: packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.scss [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    overflow-y: scroll;
+  }
+
+  &__list {
+    div {
+      height: 4rem;
```
Comment (shaundon) [inline packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.scss]: We should use a token here, just take one of the existing ones and multiply it to make something of a similar height.

File: packages/bpk-docs/src/constants/routes.js [hunk 10]
```diff
@@ -100,6 +100,7 @@ export const TOUCHABLE_NATIVE_FEEDBACK =
   '/components/touchable-native-feedback';
 export const MAP = '/components/map';
 export const CAROUSEL = '/components/carousel';
+export const INFINITE_SCROLL = '/components/infiniteScroll';
```
Comment (shaundon) [inline packages/bpk-docs/src/constants/routes.js]: This should be in kebab case to match convention, e.g. we have `/components/touchable-native-feedback`

**Comment by tiagoengel** [comment]
Addressed all feedback @shaundon and @matthewdavidson. @jamesf3rguson had looked at it before (in the original PR) but I'll ask for a new design review.

File: packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+          renderLoadingComponent={CustomLoading}
+        />
+      </FixedHeightPanel>,
+    ],
+  },
+];
+
+const blurb = [
+  <IntroBlurb>
+    This High-order component is used to wrap a list component and add the
```
Comment (shaundon) [inline packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js]: This should say `higher-order component`, not `High-order component`.

Sorry - I meant to add this to my original review and forgot.
